test/functional/test_framework/messages.py

```diff
@@ -1050,13 +1050,11 @@ def deserialize(self, f):
 
         self.nStartingHeight = struct.unpack("<i", f.read(4))[0]
 
-        if self.nVersion >= 70001:
-            # Relay field is optional for version 70001 onwards
-            try:
-                self.nRelay = struct.unpack("<b", f.read(1))[0]
-            except:
-                self.nRelay = 0
-        else:
+        # Relay field is optional for version 70001 onwards
+        # But, unconditionally check it to match behaviour in bitcoind
+        try:
+            self.nRelay = struct.unpack("<b", f.read(1))[0]
+        except:
```

Comment:
Please don't use a blanket except here, but catch the specific exception `struct.error`. This could potentially mask another error, even "KeyboardInterrupt" (ctrl-C).

Reply:
Would probably be best to remove the `except` wholesale.  https://github.com/bitcoin/bitcoin/pull/20411/files#r528499393

Reply:
This isn't new. It's simply moving the try-except out of an if block, but there's no harm in fixing it up in this PR.